### PR TITLE
Feature/variable interpolation

### PR DESCRIPTION
**Pull request by jenpet**
Addressing discussed issue #44 and a bug which did not allow to prompt variables when there is a variable on root level or on levels higher than two.

### CODE REVIEW

**Comment by jenpet** [comment]
@yields maybe you want to have a look at it. I extracted all of the interpolation relevant stuff into a new `interpolation` package which has the responsibility to interpolate tasks and the variables section.

File: interpolation/interpolation.go
Comment (yields) [inline interpolation/interpolation.go:2]: How do you feel about changing the name to `interpolate` and then the functions to `Vars()`, `Tasks()` etc.. 
Not a huge deal btw, just follows Go's naming conventions :)

File: interpolation/interpolation.go
Comment (yields) [inline interpolation/interpolation.go:62]: Should we default to the user's active shell?

**Review by yields — COMMENTED** [review]
Beautiful, I really like that new shell invocation feature, so useful :)

I left two comments, otherwise this looks great!

File: interpolation/interpolation.go
Comment (jenpet) [reply interpolation/interpolation.go:2]: You're right. I had it in the config package first and moved it afterwards. Will rename that one quickly 👍

File: interpolation/interpolation.go
Comment (jenpet) [reply interpolation/interpolation.go]: I used `$SHELL` to find a user's active shell and as a fallback it will be still `sh`. Might do the trick, what do you think?

**Comment by jenpet** [comment]
I made a separate commit with the work due to review findings but I will squash them into the previous one(s) as soon as you're happy with it so we have a clean history :-)

File: interpolation/interpolation.go
Comment (yields) [reply interpolation/interpolation.go]: sounds great to me :)

**Review by yields — APPROVED** [review]
Looks great! thanks for the contribution, I'll wait for you to squash before I merge and release.

**Comment by jenpet** [comment]
Sweet :-). I'm done so far. I got more ideas which might be useful. I'd just open a ticket for them at some point in time and create pull requests if you guys are happy with the approaches.

Thanks for releasing!

**Pull request merged**
